### PR TITLE
[15.0][ADD] Add module account_invoice_crm_tag

### PR DESCRIPTION
**Pull request by stefan-tecnativa**
TT44457 cc @Tecnativa

This module adds crm tags for invoices and refunds coming from sale_orders, and links them to its respective order.

@pedrobaeza @CarlosRoca13 please review!

### CODE REVIEW

File: account_invoice_crm_tag/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends("order_line.invoice_lines")
+    def _get_invoiced(self):
+        res = super()._get_invoiced()
+        self.invoice_ids.tags_ids = self.tag_ids
+        return res
```
Comment (CarlosRoca13) [inline account_invoice_crm_tag/models/sale_order.py]: This is not correct, you have to search the method used to create the invoices from the sale.order and see how it feels the info.

**Review by CarlosRoca13 — CHANGES_REQUESTED** [review]

File: account_invoice_crm_tag/models/sale_order.py (hunk 1 above)
Comment (pedrobaeza) [reply account_invoice_crm_tag/models/sale_order.py]: It's a `_prepare...` method in `sale.order.line` and you have to use `[(4, id)]` syntax.

Comment (stefan-tecnativa) [reply account_invoice_crm_tag/models/sale_order.py]: I saw both methods from `sale.order` and `sale.order.line`, but I wasn't sure which to use. Finally i decided to use the `sale.order` one, but still wondering what is the best decision.

File: account_invoice_crm_tag/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2023 Tecnativa - Stefan Ungureanu
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Account Invoice CRM Tag",
+    "version": "15.0.1.0.1",
```
Comment (pedrobaeza) [inline account_invoice_crm_tag/__manifest__.py]: Why this version?

```suggestion
    "version": "15.0.1.0.0",
```

File: account_invoice_crm_tag/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    "category": "Accounting",
+    "author": "Tecnativa, Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/account-invoicing",
+    "license": "AGPL-3",
+    "depends": ["account", "crm", "sale"],
```
Comment (pedrobaeza) [inline account_invoice_crm_tag/__manifest__.py]: No need of such dependencies, as the tags in the sale.order belongs to `sale`, and `sale` already depends on `account`:

```suggestion
    "depends": ["sale"],
```

File: account_invoice_crm_tag/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -0,0 +1,2 @@
+This module allows you to add crm tags to invoices coming from sales orders
```
Comment (pedrobaeza) [inline account_invoice_crm_tag/readme/DESCRIPTION.rst]: You can put manually the tags. The propagation from sales order is an extra feature that you should mention also, but not only.

File: account_invoice_crm_tag/readme/USAGE.rst [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#. Configure a sale_order, confirm it, create an invoice and open it.
```
Comment (pedrobaeza) [inline account_invoice_crm_tag/readme/USAGE.rst]: This usage should be oriented to only see and handle the tags on the invoice.

File: account_invoice_crm_tag/report/account_invoice_report.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        relation="account_move_crm_tag_rel",
+        column1="account_move_id",
+        column2="crm_tag_id",
+        string="Invoice CRM Tags",
+        readonly=True,
+    )
+
+    def _select(self):
+        select_str = super()._select()
+        return "%s, account_move_crm_tag_rel.crm_tag_id as tag_ids" % select_str
```
Comment (pedrobaeza) [inline account_invoice_crm_tag/report/account_invoice_report.py]: Seeing https://github.com/odoo/odoo/blob/15.0/addons/project/report/project_report.py, there's no need of adding it to the select.

Comment (stefan-tecnativa) [reply account_invoice_crm_tag/report/account_invoice_report.py]: Changes done, I updated the from insertion more accordingly with your reference.

Comment (chienandalu) [reply account_invoice_crm_tag/report/account_invoice_report.py]: Isn't just declaring the orm m2m field enough?

Comment (stefan-tecnativa) [reply account_invoice_crm_tag/report/account_invoice_report.py]: @chienandalu You are right, thank you

**Review by chienandalu — APPROVED** [review]
:+1:

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**